File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/Claim.ts
import { Address } from "./Address";
import { Attachment } from "./Attachment";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Money } from "./Money";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type ClaimInsurance = BackboneElement & {
    focal: boolean;
    coverage: Reference;
    sequence: number;
    identifier?: Identifier;
    preAuthRef?: string[];
    claimResponse?: Reference;
    businessArrangement?: string;
};

export type ClaimDiagnosis = BackboneElement & {
    type?: CodeableConcept[];
    sequence: number;
    diagnosis?: Reference | CodeableConcept;
    onAdmission?: CodeableConcept;
    packageCode?: CodeableConcept;
};

export type ClaimSupportingInfo = BackboneElement & {
    category: CodeableConcept;
    value?: Reference | Quantity | string | boolean | Attachment;
    reason?: CodeableConcept;
    timing?: Period | string;
    sequence: number;
    code?: CodeableConcept;
};

export type ClaimItemDetailSubDetail = BackboneElement & {
    category?: CodeableConcept;
    modifier?: CodeableConcept[];
    revenue?: CodeableConcept;
    net?: Money;
    productOrService: CodeableConcept;
    udi?: Reference[];
    programCode?: CodeableConcept[];
    factor?: number;
    sequence: number;
    quantity?: Quantity;
    unitPrice?: Money;
};

export type ClaimItemDetail = BackboneElement & {
    category?: CodeableConcept;
    modifier?: CodeableConcept[];
    revenue?: CodeableConcept;
    net?: Money;
    productOrService: CodeableConcept;
    udi?: Reference[];
    programCode?: CodeableConcept[];
    factor?: number;
    sequence: number;
    subDetail?: ClaimItemDetailSubDetail[];
    quantity?: Quantity;
    unitPrice?: Money;
};

export type ClaimItem = BackboneElement & {
    category?: CodeableConcept;
    diagnosisSequence?: number[];
    procedureSequence?: number[];
    modifier?: CodeableConcept[];
    revenue?: CodeableConcept;
    encounter?: Reference[];
    net?: Money;
    serviced?: string | Period;
    subSite?: CodeableConcept[];
    careTeamSequence?: number[];
    productOrService: CodeableConcept;
    udi?: Reference[];
    informationSequence?: number[];
    programCode?: CodeableConcept[];
    factor?: number;
    sequence: number;
    bodySite?: CodeableConcept;
    quantity?: Quantity;
    location?: Address | CodeableConcept | Reference;
    unitPrice?: Money;
    detail?: ClaimItemDetail[];
};

export type ClaimProcedure = BackboneElement & {
    udi?: Reference[];
    date?: string;
    type?: CodeableConcept[];
    sequence: number;
    procedure?: Reference | CodeableConcept;
};

export type ClaimRelated = BackboneElement & {
    claim?: Reference;
    reference?: Identifier;
    relationship?: CodeableConcept;
};

export type ClaimAccident = BackboneElement & {
    date: string;
    type?: CodeableConcept;
    location?: Address | Reference;
};

export type ClaimPayee = BackboneElement & {
    type: CodeableConcept;
    party?: Reference;
};

export type ClaimCareTeam = BackboneElement & {
    role?: CodeableConcept;
    provider: Reference;
    sequence: number;
    responsible?: boolean;
    qualification?: CodeableConcept;
};

export type Claim = DomainResource & {
    patient: Reference;
    insurance: ClaimInsurance[];
    facility?: Reference;
    diagnosis?: ClaimDiagnosis[];
    enterer?: Reference;
    supportingInfo?: ClaimSupportingInfo[];
    use: string;
    item?: ClaimItem[];
    type: CodeableConcept;
    created: string;
    procedure?: ClaimProcedure[];
    related?: ClaimRelated[];
    referral?: Reference;
    total?: Money;
    insurer?: Reference;
    fundsReserve?: CodeableConcept;
    priority: CodeableConcept;
    accident?: ClaimAccident;
    status: string;
    payee?: ClaimPayee;
    prescription?: Reference;
    billablePeriod?: Period;
    identifier?: Identifier[];
    subType?: CodeableConcept;
    provider: Reference;
    originalPrescription?: Reference;
    careTeam?: ClaimCareTeam[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Claim.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Address
from base import Attachment
from base import Period
from base import CodeableConcept
from base import Quantity
from base import DomainResource
from base import Money
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class ClaimInsurance(BackboneElement):
    focal: bool
    coverage: Reference
    sequence: int
    identifier: Optional[Identifier] = None
    pre_auth_ref: Optional[List[str]] = field(default_factory=list)
    claim_response: Optional[Reference] = None
    business_arrangement: Optional[str] = None

@dataclass(kw_only=True)
class ClaimDiagnosis(BackboneElement):
    type: Optional[List[CodeableConcept]] = field(default_factory=list)
    sequence: int
    on_admission: Optional[CodeableConcept] = None
    package_code: Optional[CodeableConcept] = None
    diagnosis_reference: Optional[Reference] = None
    diagnosis_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class ClaimSupportingInfo(BackboneElement):
    category: CodeableConcept
    value_reference: Optional[Reference] = None
    value_quantity: Optional[Quantity] = None
    timing_period: Optional[Period] = None
    value_string: Optional[str] = None
    value_boolean: Optional[bool] = None
    reason: Optional[CodeableConcept] = None
    sequence: int
    code: Optional[CodeableConcept] = None
    timing_date: Optional[str] = None
    value_attachment: Optional[Attachment] = None

@dataclass(kw_only=True)
class ClaimItemDetailSubDetail(BackboneElement):
    category: Optional[CodeableConcept] = None
    modifier: Optional[List[CodeableConcept]] = field(default_factory=list)
    revenue: Optional[CodeableConcept] = None
    net: Optional[Money] = None
    product_or_service: CodeableConcept
    udi: Optional[List[Reference]] = field(default_factory=list)
    program_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    factor: Optional[float] = None
    sequence: int
    quantity: Optional[Quantity] = None
    unit_price: Optional[Money] = None

@dataclass(kw_only=True)
class ClaimItemDetail(BackboneElement):
    category: Optional[CodeableConcept] = None
    modifier: Optional[List[CodeableConcept]] = field(default_factory=list)
    revenue: Optional[CodeableConcept] = None
    net: Optional[Money] = None
    product_or_service: CodeableConcept
    udi: Optional[List[Reference]] = field(default_factory=list)
    program_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    factor: Optional[float] = None
    sequence: int
    sub_detail: Optional[List[ClaimItemDetailSubDetail]] = field(default_factory=list)
    quantity: Optional[Quantity] = None
    unit_price: Optional[Money] = None

@dataclass(kw_only=True)
class ClaimItem(BackboneElement):
    category: Optional[CodeableConcept] = None
    diagnosis_sequence: Optional[List[int]] = field(default_factory=list)
    procedure_sequence: Optional[List[int]] = field(default_factory=list)
    location_address: Optional[Address] = None
    modifier: Optional[List[CodeableConcept]] = field(default_factory=list)
    revenue: Optional[CodeableConcept] = None
    encounter: Optional[List[Reference]] = field(default_factory=list)
    location_codeable_concept: Optional[CodeableConcept] = None
    net: Optional[Money] = None
    sub_site: Optional[List[CodeableConcept]] = field(default_factory=list)
    care_team_sequence: Optional[List[int]] = field(default_factory=list)
    product_or_service: CodeableConcept
    location_reference: Optional[Reference] = None
    udi: Optional[List[Reference]] = field(default_factory=list)
    information_sequence: Optional[List[int]] = field(default_factory=list)
    program_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    factor: Optional[float] = None
    serviced_date: Optional[str] = None
    sequence: int
    body_site: Optional[CodeableConcept] = None
    quantity: Optional[Quantity] = None
    unit_price: Optional[Money] = None
    serviced_period: Optional[Period] = None
    detail: Optional[List[ClaimItemDetail]] = field(default_factory=list)

@dataclass(kw_only=True)
class ClaimProcedure(BackboneElement):
    udi: Optional[List[Reference]] = field(default_factory=list)
    date: Optional[str] = None
    type: Optional[List[CodeableConcept]] = field(default_factory=list)
    sequence: int
    procedure_reference: Optional[Reference] = None
    procedure_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class ClaimRelated(BackboneElement):
    claim: Optional[Reference] = None
    reference: Optional[Identifier] = None
    relationship: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class ClaimAccident(BackboneElement):
    date: str
    type: Optional[CodeableConcept] = None
    location_address: Optional[Address] = None
    location_reference: Optional[Reference] = None

@dataclass(kw_only=True)
class ClaimPayee(BackboneElement):
    type: CodeableConcept
    party: Optional[Reference] = None

@dataclass(kw_only=True)
class ClaimCareTeam(BackboneElement):
    role: Optional[CodeableConcept] = None
    provider: Reference
    sequence: int
    responsible: Optional[bool] = None
    qualification: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class Claim(DomainResource):
    patient: Reference
    insurance: list[ClaimInsurance] = field(default_factory=list)
    facility: Optional[Reference] = None
    diagnosis: Optional[List[ClaimDiagnosis]] = field(default_factory=list)
    enterer: Optional[Reference] = None
    supporting_info: Optional[List[ClaimSupportingInfo]] = field(default_factory=list)
    use: str
    item: Optional[List[ClaimItem]] = field(default_factory=list)
    type: CodeableConcept
    created: str
    procedure: Optional[List[ClaimProcedure]] = field(default_factory=list)
    related: Optional[List[ClaimRelated]] = field(default_factory=list)
    referral: Optional[Reference] = None
    total: Optional[Money] = None
    insurer: Optional[Reference] = None
    funds_reserve: Optional[CodeableConcept] = None
    priority: CodeableConcept
    accident: Optional[ClaimAccident] = None
    status: str
    payee: Optional[ClaimPayee] = None
    prescription: Optional[Reference] = None
    billable_period: Optional[Period] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    sub_type: Optional[CodeableConcept] = None
    provider: Reference
    original_prescription: Optional[Reference] = None
    care_team: Optional[List[ClaimCareTeam]] = field(default_factory=list)